components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementUtil.java

```diff
@@ -439,44 +499,20 @@ public static boolean addFilesystemPolicy(PolicyDTO policyDTO,
         policyObj = getPolicy(policyDTO.getPolicy());
 
         if (policyObj != null) {
-            PAPPolicyStore policyStore = new PAPPolicyStore(registry);
             policyAdmin = new PAPPolicyStoreManager();
             policyDTO.setPolicyId(policyObj.getId().toASCIIString());
             policyDTO.setActive(true);
 
-            if (isPolicyExists(policyDTO.getPolicyId(), registry)) {
+            if (isPolicyExists(policyDTO.getPolicyId())) {
                 return false;
             }
 
             policyDTO.setPromote(promote);
-            PolicyVersionManager versionManager = EntitlementAdminEngine.getInstance().getVersionManager();
-            try {
-                String version = versionManager.createVersion(policyDTO);
-                policyDTO.setVersion(version);
-            } catch (EntitlementException e) {
-                log.error("Policy versioning is not supported", e);
-            }
             policyAdmin.addOrUpdatePolicy(policyDTO);
 
-            PAPPolicyStoreReader reader = new PAPPolicyStoreReader(policyStore);
-            policyDTO = reader.readPolicyDTO(policyDTO.getPolicyId());
-
-            if (Boolean.parseBoolean(System.getProperty(ENHANCED_XACML_LOADING_SYSTEM_PROPERTY)) && promote) {
```

Comment:
Removing `Boolean.parseBoolean(System.getProperty(ENHANCED_XACML_LOADING_SYSTEM_PROPERTY)` intentional?

Reply:
According to the usage, removing this doesn't seem to have much impact, since the flow works by publishing the policy to pdp.